feat(core): add `prefixMode` option to embeddables (#6113)
mikro-orm/mikro-orm: docs/docs/embeddables.md

  </TabItem>
</Tabs>

You can also decide more precisely how the column name is determined with an explicit prefix. With the example below:

> - An `absolute` mode sets the prefix at the begining of the **column**, naming them `addr_city`, `addr_street`, ...
> - A `relative` mode **concatenates** the prefix with its parent's prefix (if any), naming them `contact_addr2_city`, `contact_addr2_street`, ...

Reviewer: ```suggestion
- `absolute` mode (default) sets the prefix at the beginning of the **column**, naming them `addr_city`, `addr_street`, ...
- `relative` mode **concatenates** the prefix with its parent's prefix (if any), naming them `contact_addr2_city`, `contact_addr2_street`, ...

:::warning

The default value of `prefixMode` will change in v7 to `relative`.

:::
```

  </TabItem>
</Tabs>

> You can also decide more precisely how the column name is determined with an explicit prefix.
> With the example below:

Reviewer: ```suggestion
You can also decide more precisely how the column name is determined with an explicit prefix. With the example below:
```

  address2!: Address;
}
@Entity()
export class User {
  @Embedded(() => Contact)
  contact!: Contact;
}
```

  </TabItem>
  <TabItem value="ts-morph">

```ts title="./entities/User.ts"
@Embeddable()
class Contact {
  @Embedded({ prefix: 'addr_', prefixBehavior: 'absolute' })
  address!: Address;
  @Embedded({ prefix: 'addr2_', prefixBehavior: 'relative' })
  address2!: Address;
}
@Entity()
export class User {
  @Embedded()
  contact!: Contact;
}
```

  </TabItem>
  <TabItem value="entity-schema">

```ts title="./entities/User.ts"
class Contact {
  address!: Address;
  address2!: Address;
}
export class User {
  id!: number;
  contact!: Contact;
}

export const ContactSchema = new EntitySchema({
  class: Contact,
  embeddable: true,
  properties: {
    address: { kind: 'embedded', entity: 'Address', prefix: 'addr_', prefixBehavior: 'absolute' },
    address2: { kind: 'embedded', entity: 'Address', prefix: 'addr2_', prefixBehavior: 'relative' },
  },
});
export const UserSchema = new EntitySchema({
  class: User,
  properties: {
    id: { primary: true, type: 'number' },
    contact: { kind: 'embedded', entity: 'Contact' },
  },
});

> 

Reviewer: there are many code style issues in this block, pay more attention to blank lines (one between every class/interface, but no blank line at the end of the code block

```

  </TabItem>
</Tabs>

The default behavior can be defined in the ORM configuration:

```ts

> MikroORM.init({ embeddable: { prefixBehavior: "absolute" }})

Reviewer: be consistent, code style in docs should be the same as in code

```suggestion
MikroORM.init({ embeddables: { prefixBehavior: 'absolute' }})
```
